### PR TITLE
update code for zk summit

### LINKED ISSUE #2: Update User model and registration flow to work with Webauthn
Update the User model, integrate Webauthn account creation flow

### LINKED ISSUE #3: Update person tap to fetch socials, remove share flow from tap
Socials should be stored on server for sharing, tap should fetch socials, remove share page from connection flow

### LINKED ISSUE #9: Remove all non-applicable code from nfc-denver
- [ ] item store
- [ ] buidl
- [ ] pending tab
- [ ] wallet connection
- [ ] self vs server custody

### LINKED ISSUE #11: Update proofs tab to only have one requirement which generates a QR code upon completion

### LINKED ISSUE #18: Ensure profile updates propagate socials

### LINKED ISSUE #20: User should fetch profile info from server on login
Fetch socials from server upon login

### LINKED ISSUE #24: Find solution for getting user signatures

### PR DESCRIPTION
**Pull request by AndrewCLu**
- [x] Update database models for zk summit
- [x] Webauthn registration and login
- [x] Fetch user socials upon tap
- [x] Remove share flow from user tap
- [x] Update proofs to only display one requirement
- [x] Update proof generation and QR code to work with new proof structure
- [x] Fix profile update code to propagate socials
- [x] Find solution for getting user signatures from tap
- [x] Remove irrelevant code
- [x] Set up deployment

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,14 +1,9 @@
 DATABASE_PRISMA_URL="postgresql://postgres:postgres@localhost:5432/test?schema=public"
 DATABASE_URL_NON_POOLING="postgresql://postgres:postgres@localhost:5432/test?schema=public"
 IYK_EMAIL_API_SECRET="1234567890"
```
Comment (RiverRuby) [inline .env.example]: We don't need this right?

File: src/lib/client/localStorage/profile.ts [hunk 2]
```diff
@@ -9,11 +9,8 @@ export const PROFILE_STORAGE_KEY = "profile";
...
 export type Profile = {
   displayName: string;
-  email: string;
   encryptionPublicKey: string;
   signaturePublicKey: string;
-  wantsServerCustody: boolean;
-  allowsAnalytics: boolean;
   twitterUsername?: string;
   telegramUsername?: string;
   farcasterUsername?: string;
```
Comment (RiverRuby) [inline src/lib/client/localStorage/profile.ts]: Any reason this was kept in profile?

File: src/lib/client/quests.ts [hunk 3]
```diff
@@ -17,32 +17,44 @@ export const computeNumRequirementsSatisfied = (args: {
...
+  }
 
-  const numLocationRequirementsSatisfied = locationRequirements.filter(
-    (requirement) =>
-      computeNumRequirementSignatures({
-        publicKeyList: locationPublicKeys,
-        locationRequirement: requirement,
-      }) >= requirement.numSigsRequired
-  ).length;
+  // const numUserRequirementsSatisfied = userRequirements.filter(
```
Comment (RiverRuby) [inline src/lib/client/quests.ts]: Why is this commented instead of just deleted? prefer delete

File: src/lib/schema/schema.ts [hunk 4]
```diff
@@ -66,13 +61,6 @@ export const ProfileSchema = object({
     })
     .trim()
     .optional(),
-  farcasterUsername: string()
```
Comment (RiverRuby) [inline src/lib/schema/schema.ts:69]: And then in other places farcaster is kept

File: src/lib/shared/backup.ts [hunk 5]
```diff
@@ -8,8 +8,8 @@ import {
 const ENCRYPTION_KEY_LENGTH = 32; // AES-256 keys are 32 bytes
 const IV_LENGTH = 12; // 12 byte IV for GCM mode
 const SALT = "zk-is-cool"; // We want deterministic encryption based on email and password
-const ITERATIONS = 100000; // Number of iterations for PBKDF2
-const DIGEST = "sha512"; // Digest method for PBKDF2
+const ITERATIONS = 1000; // Number of iterations for PBKDF2
```
Comment (RiverRuby) [inline src/lib/shared/backup.ts:11]: lfg, fuck security

File: src/pages/_app.tsx [hunk 6]
```diff
@@ -14,7 +14,6 @@ import { StateMachineProvider } from "little-state-machine";
 import type { AppProps } from "next/app";
 import { useEffect, useState } from "react";
 import { toast, Toaster } from "sonner";
-import PlausibleProvider from "next-plausible";
```
Comment (RiverRuby) [inline src/pages/_app.tsx:17]: What will we use instead of Plausible?

File: src/pages/_document.tsx [hunk 7]
```diff
@@ -1,20 +1,19 @@
...
 export default function Document() {
   return (
     <Html lang="en">
-      <title>BUIDLQuest</title>
+      <title>ZK Summit 11</title>
       <Head>
         <meta
           name="description"
-          content="Connect with ETHDenver attendees, engage with sponsors, and unlock unique experiences by tapping NFC chips."
+          content="Connect with ZK Summit 11 attendees by tapping NFC chips."
```
Comment (RiverRuby) [inline src/pages/_document.tsx]: Can this more closely mirror the copy on the cards?

File: src/pages/api/backup.ts [hunk 8]
```diff
@@ -75,80 +74,41 @@ export default async function handler(
...
-    // Retrieve user and check wantsServerCustody matches
+    // Retrieve user
     const user = await prisma.user.findUnique({
       where: { id: userId },
     });
     if (!user) {
       return res.status(404).json({ error: "User not found" });
     }
 
-    // For safety, check that wantsServerCustody matches
```
Comment (RiverRuby) [inline src/pages/api/backup.ts:94]: Default self custody is fucking lit

File: src/pages/api/quest/submit_proof.tsx [hunk 9]
```diff
@@ -48,32 +48,32 @@ export default async function submitProofHandler(
...
-          (message) => message.senderId !== message.recipientId
-        ).length;
-        if (outboundTotal < quest.userTapReq) {
-          return res.status(400).json({ error: "Have tapped enough people" });
-        }
-      } else {
-        return res.status(400).json({ error: "No outbound taps" });
-      }
-    }
+    // if (quest.userTapReq !== null) {
```
Comment (RiverRuby) [inline src/pages/api/quest/submit_proof.tsx]: Let's just delete?

File: src/pages/qr/[id].tsx [hunk 10]
```diff
@@ -145,42 +102,6 @@ const QRPage = () => {
...
-          }),
-        });
-
-        if (!response.ok) {
-          throw new Error("Received bad status code from server");
-        }
-      } catch (error) {
-        console.error("Failed to send item redeemed message: ", error);
-      }
       toast.success("Successfully redeemed item for user!");
```
Comment (RiverRuby) [inline src/pages/qr/[id].tsx]: redeemed item?

**Review by RiverRuby — COMMENTED** [review]
I mostly skimmed, but I figure we can test this stuff in more detail over the week instead of right now. Overall LGTM, it is so fucking satisfying to rip out all the item store related stuff LOLLLLLLLLLLL

can do another review after signature stuff is added

File: src/pages/_app.tsx (hunk 6 above)
Comment (AndrewCLu) [reply src/pages/_app.tsx:17]: matomo

File: src/pages/api/register/create_account.ts [hunk 11]
```diff
@@ -124,12 +140,21 @@ export default async function handler(
...
-      twitter,
-      telegram,
+      twitter: parsedTwitter,
+      telegram: parsedTelegram,
       bio,
     },
   });
 
+  // Create chip key
+  await prisma.chipKey.create({
```
Comment (RiverRuby) [inline src/pages/api/register/create_account.ts:150]: I thought we would make the signing keys before hand so we can pre-assign groups?

**Review by RiverRuby — APPROVED** [review]
LGTM if tests work!

File: src/pages/api/register/create_account.ts (hunk 11 above)
Comment (RiverRuby) [reply src/pages/api/register/create_account.ts:150]: we can keep this in for now to quickly test and iterate but yeah i think i misunderstood

Comment (AndrewCLu) [reply src/pages/api/register/create_account.ts:150]: this is for testing, see: https://github.com/cursive-team/zk-summit/issues/28

**Comment by AndrewCLu** [comment]
Closes #2 #3 #9 #11 #18 #20 #24

**Pull request merged**